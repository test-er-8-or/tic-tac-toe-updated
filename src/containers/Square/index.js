import Square from '../../components/Square'
import { connect } from 'react-redux'
import { getMoves } from '../../state/selectors'
import getPlayer from '../../utilities/getPlayer'
import { squareClicked } from '../../state/actions'

function mapStateToProps (state, { index }) {
  const moves = getMoves(state)

  return {
    player: getPlayer(index, moves)
  }
}

function mapDispatchToProps (dispatch, { index }) {
  return {
    handleClick: () => dispatch(squareClicked(index))
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Square)
