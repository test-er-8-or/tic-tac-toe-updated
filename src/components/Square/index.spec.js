import * as React from 'react'

import Square from './'
import { shallow } from 'enzyme'

describe('components:Square', () => {
  it('matches the snapshot for player O in the top left square', () => {
    expect(
      toJson(shallow(<Square player='o' index={0} />).dive())
    ).toMatchSnapshot()
  })

  it('matches the snapshot for player X in the top left square', () => {
    expect(
      toJson(shallow(<Square player='x' index={0} />).dive())
    ).toMatchSnapshot()
  })

  it('matches the snapshot for player X in the top right square', () => {
    expect(
      toJson(shallow(<Square player='x' index={2} />).dive())
    ).toMatchSnapshot()
  })

  it('matches the snapshot for player X in the bottom left square', () => {
    expect(
      toJson(shallow(<Square player='x' index={6} />).dive())
    ).toMatchSnapshot()
  })

  it('matches the snapshot for player X in the bottom right square', () => {
    expect(
      toJson(shallow(<Square player='x' index={8} />).dive())
    ).toMatchSnapshot()
  })
})
