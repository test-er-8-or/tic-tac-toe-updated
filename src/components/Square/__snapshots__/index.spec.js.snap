// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components:Square matches the snapshot for player O in the top left square 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  border-color: hsla(0, 0%, 0%, 0.2);
  border-style: solid;
  border-width: 0 ",
          [Function],
          "
    ",
          [Function],
          " 0;
  color: ",
          [Function],
          ";
  font-size: 16vh;
  font-weight: bold;
  line-height: 20vh;
  text-align: center;
  text-transform: uppercase;
",
        ],
      },
      "displayName": "StyledSquare",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  index={0}
  player="o"
>
  o
</StyledComponent>
`;

exports[`components:Square matches the snapshot for player X in the bottom left square 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  border-color: hsla(0, 0%, 0%, 0.2);
  border-style: solid;
  border-width: 0 ",
          [Function],
          "
    ",
          [Function],
          " 0;
  color: ",
          [Function],
          ";
  font-size: 16vh;
  font-weight: bold;
  line-height: 20vh;
  text-align: center;
  text-transform: uppercase;
",
        ],
      },
      "displayName": "StyledSquare",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  index={6}
  player="x"
>
  x
</StyledComponent>
`;

exports[`components:Square matches the snapshot for player X in the bottom right square 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  border-color: hsla(0, 0%, 0%, 0.2);
  border-style: solid;
  border-width: 0 ",
          [Function],
          "
    ",
          [Function],
          " 0;
  color: ",
          [Function],
          ";
  font-size: 16vh;
  font-weight: bold;
  line-height: 20vh;
  text-align: center;
  text-transform: uppercase;
",
        ],
      },
      "displayName": "StyledSquare",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  index={8}
  player="x"
>
  x
</StyledComponent>
`;

exports[`components:Square matches the snapshot for player X in the top left square 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  border-color: hsla(0, 0%, 0%, 0.2);
  border-style: solid;
  border-width: 0 ",
          [Function],
          "
    ",
          [Function],
          " 0;
  color: ",
          [Function],
          ";
  font-size: 16vh;
  font-weight: bold;
  line-height: 20vh;
  text-align: center;
  text-transform: uppercase;
",
        ],
      },
      "displayName": "StyledSquare",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  index={0}
  player="x"
>
  x
</StyledComponent>
`;

exports[`components:Square matches the snapshot for player X in the top right square 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  border-color: hsla(0, 0%, 0%, 0.2);
  border-style: solid;
  border-width: 0 ",
          [Function],
          "
    ",
          [Function],
          " 0;
  color: ",
          [Function],
          ";
  font-size: 16vh;
  font-weight: bold;
  line-height: 20vh;
  text-align: center;
  text-transform: uppercase;
",
        ],
      },
      "displayName": "StyledSquare",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  index={2}
  player="x"
>
  x
</StyledComponent>
`;
